oauth2_http/javatests/com/google/auth/oauth2/ExternalAccountCredentialsTest.java

```diff
@@ -282,6 +325,104 @@ public void getRequestMetadata_withQuotaProjectId() throws IOException {
     assertEquals("quotaProjectId", requestMetadata.get("x-goog-user-project").get(0));
   }
 
+  @Test
+  public void validateTokenUrl_validUrls() {
+    List<String> validUrls =
+        Arrays.asList(
+            "https://sts.googleapis.com",
+            "https://us-east-1.sts.googleapis.com",
+            "https://US-EAST-1.sts.googleapis.com",
+            "https://sts.us-east-1.googleapis.com",
+            "https://sts.US-WEST-1.googleapis.com",
+            "https://us-east-1-sts.googleapis.com",
+            "https://US-WEST-1-sts.googleapis.com",
+            "https://us-west-1-sts.googleapis.com/path?query");
+
+    for (String url : validUrls) {
+      ExternalAccountCredentials.validateTokenUrl(url);
+    }
+  }
+
+  @Test
+  public void validateTokenUrl_invalidUrls() {
+    List<String> invalidUrls =
+        Arrays.asList(
+            "https://iamcredentials.googleapis.com",
+            "sts.googleapis.com",
+            "https://",
+            "https://us-eas\\t-1.sts.googleapis.com",
+            "https:/us-east-1.sts.googleapis.com",
+            "https://US-WE/ST-1-sts.googleapis.com",
```

Comment:
add a test case where some subdomain goes at the end? like https://us-east-1.sts.googleapis.com.evil.com